resolver/custom_dns_resolver.go

```diff
@@ -120,26 +137,45 @@ func (r *CustomDNSResolver) processRequest(request *model.Request) *model.Respon
 		}
 	}
 
-	return nil
+	logger.WithField("next_resolver", Name(r.next)).Trace("go to next resolver")
+	forwardResponse, _ := r.next.Resolve(ctx, request)
```

Comment:
We need to return this error, can you adjust the function signatures? `processCNAME` will need to change too.